lib/lib_install.php

```diff
@@ -41,14 +41,14 @@ function checkRequirements(string $dbType = ''): array {
 	$json = function_exists('json_encode');
 	$mbstring = extension_loaded('mbstring');
 	// @phpstan-ignore-next-line
-	$data = DATA_PATH && touch(DATA_PATH . '/index.html');	// is_writable() is not reliable for a folder on NFS
+	$data = DATA_PATH && @touch(DATA_PATH . '/index.html');	// is_writable() is not reliable for a folder on NFS
```

Comment:
dont use '@' manage false return

Reply:
I added it on purpose....
While I checked this and had the file not available, I got a fatal error

Reply:
catch the fatal...

Reply:
https://www.php.net/manual/en/function.touch.php

touch() return an boolean 
```
if (touch($filename)) {
    echo $filename . ' modification time has been changed to present time';
} else {
    echo 'Sorry, could not change modification time of ' . $filename;
}
```

Reply:
Strange. I cannot reproduce the fatal error

Reply:
I disagree about not using `@` as a general rule. Did you try your own suggestion @ColonelMoutarde ?

Here is an example:

```php
<?php
try {
	$ok = touch('/blabla/truc');
	echo json_encode($ok);
} catch (Exception $ex) {
	echo 'Handled exception ' . $ex;
}
```

It outputs:

```
PHP Warning:  touch(): Unable to create file /blabla/truc because No such file or directory in /home/alexandre/GitHub/FreshRSS/tests/test-touch.php on line 3
```

Reply:
https://stackoverflow.com/questions/1241728/can-i-try-catch-a-warning

Reply:
Exactly, one cannot *catch the fatal* or *manage false return*. Approaches other than `@` that I have seen so far are verbose, heavy, and error prone.

Reply:
Agreed, it's less readable at best, though a PHP newbie might not be familiar with `@` so in that sense a try/catch is potentially mildly preferable, though I doubt it. :-)

Reply:
transforming warnings into exceptions is the method used by modern frameworks (laravel, symfony, yii...).
```
set_error_handler(function ($severity, $message, $file, $line) {
    throw new \ErrorException($message, $severity, $severity, $file, $line);
});

$result = file_get_contents($url);
```
It works very well

Reply:
Not for now, sorry. And we are not interested in catching any exception in this case, just suppressing the warning, as we are properly handling the different return codes.

Reply:
more about the `touch()`: `If the file does not exist, it will be created.` It will create an empty file. https://www.php.net/manual/en/function.touch.php
But: If the process has no rights to create the file, then it will throw the fatal error. 
the information `file does not exists` only would be enough here.

I would go forward with the `@` to suppress the error.

Reply:
To check the permissions of a file via a PHP script, i.e. dynamically, we can use the PHP function fileperms() which returns the permissions for a file.

This function will return the permissions of a file in numerical form. We can then convert the result into octal form with the decoct() function to obtain the representation of the permissions according to our three numbers.

Note that fileperms() can return additional information depending on the system used. The last three digits returned are the permissions.

We can also use the functions is_readable() and is_writable() which will respectively determine if the file can be read and written to. These functions will return true if it is the case or false if it is not, which makes them very practical to use within a condition.

Reply:
see `lib_install.php` line 122:

`function file_permissions(string $path): string {`

Reply:
I added now a `file_exists()`

New code line:
`$data = DATA_PATH && file_exists(DATA_PATH . '/index.html') && touch(DATA_PATH . '/index.html');`

Reply:
Nope, `is_writable()` is what we used before (please check git line log, and it is even in comment there) and it does not work for instance on a folder on NFS